app/models/user.rb

```diff
@@ -0,0 +1,35 @@
+class User < ApplicationRecord
+  attr_accessor :remember_token
+  before_create :remember_token
+  before_save { self.email = email.downcase }
+  validates :name, presence: true, length: { maximum: 30 }
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i.freeze
+  validates :email, presence: true, length: { maximum: 255 },
+                    format: { with: VALID_EMAIL_REGEX },
+                    uniqueness: { case_sensitive: false }
+  validates :password, length: { minimum: 6 }, presence: true
+  has_secure_password
+  has_many :posts
+
+  def self.digest(string)
+    cost = ActiveModel::SecurePassword.min_cost ? BCrypt::Engine::MIN_COST : BCrypt::Engine.cost
+    BCrypt::Password.create(string, cost: cost)
+  end
+
+  def self.new_token
+    SecureRandom.urlsafe_base64
+  end
+
+  def forget
+    update_attribute(:remember_digest, nil)
+  end
+
+  def remember
+    self.remember_token = User.new_token
+    update_attribute(:remember_digest, User.digest(remember_token))
+  end
```

Comment:
If you follow the instructions given up, you will update it with only the last lines